web3-react/src/libs/components/Option.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useCallback, useMemo } from 'react'
+
+import { ConnectionType, getConnection, tryActivateConnector, tryDeactivateConnector } from '../connections'
+
+export const Option = ({
+  isActive,
+  activeConnectionType,
+  connectionType,
+  onActivate,
+  onDeactivate,
+}: {
+  isActive: boolean
+  activeConnectionType: ConnectionType | null
+  connectionType: ConnectionType
+  onActivate: (connectionType: ConnectionType) => void
+  onDeactivate: (connectionType: null) => void
+}) => {
+  const isOptionActive = useMemo(
+    () => isActive && activeConnectionType === connectionType,
+    [isActive, activeConnectionType, connectionType]
+  )
+
+  const isOtherOptionActive = useMemo(
+    () => isActive && activeConnectionType !== connectionType && activeConnectionType !== ConnectionType.NETWORK,
+    [isActive, activeConnectionType, connectionType]
+  )
```

Comment:
Quick question on conditionals here:

- Can `activeConnectionType` have a value when `isActive` is false?
- Why do we need to check for  `activeConnectionType !== ConnectionType.NETWORK`? Isn't `activeConnectionType !== connectionType` just fine?

The reason I am asking is this: I feel a bit overwhelmed by these conditionals. I feel like a simple `isOptionActive` of value `isActive && activeConnectionType === connectionType` would do.

Then, `isOtherOptionActive` is as simple as `isActive && !isOptionActive`

Reply:
Another thing (this is more of a nitpick, but since we're building an example, I feel that we need to encourage right encapsulation):

I would personally pass `disabled` instead of `isActive` and the entire `isOtherOptionActive` logic inside this component. I feel like `Option` component setting its internal appearance based on other components' state breaks encapsulation and would be better suited in a parent component.

Again, it's more of a nit-pick, so I am leaving final decision up to you.

Reply:
Make the conditionals simpler (I was missing a case), and also change the param